sdks/python/apache_beam/io/iobase.py

```diff
@@ -1456,10 +1473,17 @@ def try_split(self, fraction_of_remainder):
         residual_weight = self._weight - primary_weight
         # Update self._weight to primary weight
         self._weight = primary_weight
-        return (SourceBundle(primary_weight, self._source, start_pos,
-                             split_pos),
-                SourceBundle(residual_weight, self._source, split_pos,
-                             stop_pos))
+        return (
```

Comment:
You could add a split method to the _SDFBoundedSourceRestriction that would contain most of the logic of this method.